### DISCUSSION TITLE
Getting Null Values, instead of parsed values

### DISCUSSION THREAD

**Discussion by MV45716** [vectordotdev/vector#17558] [Q&A]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Use Case: We are designing a pipeline to ingest a json object that will output/post a multiline string to an api. 

We are able to create the string and add it successfully to a json object however when we try to replace the json document entirely with the string the variables are returning null values.

VRL Playground link: [Link for Playground code](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLm1ldHJpY05hbWUsIGVycj0gdG9fc3RyaW5nKC5tZXRyaWNuYW1lKVxuICAgICAgaWYgZXJyICE9IG51bGwge1xuICAgICAgICBsb2coXCJFcnJvciBvbiBtZXRyaWNuYW1lOiBcIiArIGVyciwgbGV2ZWw6IFwiZXJyb3JcIilcbiAgICAgIH0gXG4gICAgICAuc2xvSUQsIGVycj0gdG9fc3RyaW5nKC5zbG9pZClcbiAgICAgIGlmIGVyciAhPSBudWxsIHtcbiAgICAgICAgbG9nKFwiRXJyb3Igb24gc2VydmljZWlkOiBcIiArIGVyciwgbGV2ZWw6IFwiZXJyb3JcIilcbiAgICAgIH1cbiAgICAgIC5zdGF0dXM9XCJBUFBST1ZFRFwiXG4gICAgICAubWV0cmljVj0gLnNlcmllc1swXS5wb2ludGxpc3RbMF1cbiAgICAgIC5tZXRyaWNWYWx1ZSwgZXJyPXRvX3N0cmluZygubWV0cmljVlsxXSlcbiAgICAgIGlmIGVyciAhPSBudWxsIHtcbiAgICAgICAgbG9nKFwiRXJyb3Igb24gcG9pbnRsaXN0IHZhbHVlOiBcIiArIGVyciwgbGV2ZWw6IFwiZXJyb3JcIilcbiAgICAgIH0gXG4gICAgICAuPVwiIyBUWVBFIFwiKy5tZXRyaWNOYW1lK1wiIGdhdWdlXG4jIEhFTFAgXCIrLm1ldHJpY05hbWUrXCIgXCIrcmVwbGFjZSgubWV0cmljTmFtZSwgXCJfXCIsIFwiIFwiKStcIiBnYXVnZVxuXCIrLm1ldHJpY05hbWUrXCJ7c2xvSWQ9XFxcIlwiKy5zbG9JRCtcIlxcXCIsc3RhdHVzPVxcXCJcIisuc3RhdHVzK1wiXFxcIn0gXCIrLm1ldHJpY1ZhbHVlK1wiXG4jIEVPRlxuXCIiLCJldmVudCI6eyJzdGF0dXMiOiJvayIsInJlc190eXBlIjoidGltZV9zZXJpZXMiLCJyZXNwX3ZlcnNpb24iOjEsInF1ZXJ5Ijoic3VtOlBpcGVsaW5lX0V2ZW50X0NvdW50e3NlcnZpY2UuZW52aXJvbm1lbnQ6dGVzdH0uYXNfY291bnQoKSIsImZyb21fZGF0ZSI6MTY4MzA2NDY4MDAwMCwidG9fZGF0ZSI6MTY4MzA2NDk4MDAwMCwic2VyaWVzIjpbeyJ1bml0IjpudWxsLCJxdWVyeV9pbmRleCI6MCwiYWdnciI6InN1bSIsIm1ldHJpYyI6IlBpcGVsaW5lX0V2ZW50X0NvdW50IiwidGFnX3NldCI6W10sImV4cHJlc3Npb24iOiJzdW06UGlwZWxpbmVfRXZlbnRfQ291bnR7c2VydmljZS5lbnZpcm9ubWVudDp0ZXN0fS5hc19jb3VudCgpIiwic2NvcGUiOiJzZXJ2aWNlLmVudmlyb25tZW50OnRlc3QiLCJpbnRlcnZhbCI6MiwibGVuZ3RoIjoxLCJzdGFydCI6MTY4MzA2NDcwMDAwMCwiZW5kIjoxNjgzMDY0NzAxMDAwLCJwb2ludGxpc3QiOltbMTY4MzA2NDcwMDAwMCw3Ni44XV0sImRpc3BsYXlfbmFtZSI6IlBpcGVsaW5lX0V2ZW50X0NvdW50IiwiYXR0cmlidXRlcyI6e319XSwidmFsdWVzIjpbXSwidGltZXMiOltdLCJtZXNzYWdlIjoiIiwiZ3JvdXBfYnkiOltdLCJzbG9pZCI6IjAzM2RmMGZkOWM0MDQwYjk4ZDk0YTVlYzMyNWZkM2U5IiwibWV0cmljbmFtZSI6ImF2YWlsYWJpbGl0eV9zbG9fdG90YWxfY291bnQifSwiaXNfanNvbmwiOmZhbHNlfQ%3D%3D)

Vector transformation config:  refer config section.

console output

```
# TYPE  gauge
# HELP   gauge
{sloId="",status="APPROVED"} 
# EOF
```

Expected output:

```
# TYPE availability_slo_total_count gauge
# HELP availability_slo_total_count availability slo total count gauge
availability_slo_total_count{sloId="033df0fd9c4040b98d94a5ec325fd3e9",status="APPROVED"} 76.8
# EOF

```

### Configuration

```text
transforms:
  parse_dd_api_data:
    type: remap
    inputs: 
      - dd_api_data
    source: |-
      .metricName, err= to_string(.metricname)
      if err != null {
        log("Error on metricname: " + err, level: "error")
      } 
      .sloID, err= to_string(.sloid)
      if err != null {
        log("Error on serviceid: " + err, level: "error")
      } 
      .status="APPROVED"
      .metricV= .series[0].pointlist[0]
      .metricValue, err=to_string(.metricV[1])
      if err != null {
        log("Error on pointlist value: " + err, level: "error")
      } 
      .="# TYPE "+.metricName+" gauge
      # HELP "+.metricName+" "+replace(.metricName, "_", " ")+" gauge
      "+.metricName+"{sloId=\""+.sloID+"\",status=\""+.status+"\"} "+.metricValue+"
      # EOF
     "
sinks:
  console_output:
    type: console
    inputs:
      - parse_dd_api_data
    encoding:
      codec: text
```

### Version

vector-0.30.0-1.x86_64.rpm

### Example Data

```
{
	"status": "ok",
	"res_type": "time_series",
	"resp_version": 1,
	"query": "sum:Pipeline_Event_Count{service.environment:test}.as_count()",
	"from_date": 1683064680000,
	"to_date": 1683064980000,
	"series": [
		{
			"unit": null,
			"query_index": 0,
			"aggr": "sum",
			"metric": "Pipeline_Event_Count",
			"tag_set": [],
			"expression": "sum:Pipeline_Event_Count{service.environment:test}.as_count()",
			"scope": "service.environment:test",
			"interval": 2,
			"length": 1,
			"start": 1683064700000,
			"end": 1683064701000,
			"pointlist": [
				[
					1683064700000,
					76.8
				]
			],
			"display_name": "Pipeline_Event_Count",
			"attributes": {}
		}
	],
	"values": [],
	"times": [],
	"message": "",
	"group_by": [],
	"sloid": "033df0fd9c4040b98d94a5ec325fd3e9",
	"metricname": "availability_slo_total_count"
}
```

**Comment by neuronull** [comment]
Hello,

The VRL playground event data provides a means to give the VRL program anything you want.

When you get to an actual vector pipeline though, the event data is probably different from what you provided in that example, because we process the raw incoming data in the source.

Did you try piping directly to the console (without the transform), and inspecting the result?

Probably what is going on here is the fields you are looking for are behind the `.message` , if you are ingesting log events.
